charmcraft/linters.py

```diff
@@ -264,6 +264,10 @@ def run(self, basedir: pathlib.Path) -> str:
             self.text = f"The metadata.yaml file is missing the following attribute(s): {missing}."
             return self.Result.ERRORS
 
+        if "series" in metadata:
+            self.text = "The metadata.yaml file contains deprecated attribute: series."
+            return self.Result.ERRORS
```

Comment:
This will prevent the charm from being packed, even though it's still a valid charm on juju 2.x and 3.x.

We unfortunately can't make that change for existing bases. However, with the appropriate documentation change, we can do:

```suggestion
            self.text = "The metadata.yaml file contains the deprecated attribute: series. This attribute will be rejected starting in Juju 4.0."
            return self.Result.WARNINGS
```

If you're willing to write the extra code necessary, we can have it error for all future bases. This would mean checking `manifest.yaml` and if it contains only currently supported bases, returning the warning, but returning an error if any base is not in the current set of supported bases.

This would mean setting the result level to warning for the following bases:

- Ubuntu 18.04, 20.04, 22.04, 23.04, 23.10
- CentOS 7
- Almalinux 9

and letting it error out for other bases. 

The test case could then check that a manifest.yaml file with a `bases` section like:

```
bases:
- name: ubuntu
  channel: "18.04"
- name: ubuntu
  channel: "20.04"
- name: ubuntu
  channel: "22.04"
- name: ubuntu
  channel: "23.04"
- name: ubuntu
  channel: "23.10"
  architectures:
  - riscv64
- name: centos
  channel: "7"
- name: almalinux
  channel: "9"
```

would receive a warning, but the following would receive an error:

```
bases:
- name: ubuntu
  channel: "24.04"
```

as would:

```
bases:
- name: ubuntu
  channel: "18.04"
- name: ubuntu
  channel: "20.04"
- name: ubuntu
  channel: "22.04"
- name: ubuntu
  channel: "23.04"
- name: ubuntu
  channel: "23.10"
  architectures:
  - riscv64
- name: centos
  channel: "7"
- name: almalinux
  channel: "9"
- name: ubuntu
  channel: "24.04"
  architectures: [riscv64]
```